pkg/authentication/scram/scram_enabler.go

```diff
@@ -0,0 +1,56 @@
+package scram
+
+import (
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/automationconfig"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/util/contains"
+)
+
+const (
+	scram256                              = "SCRAM-SHA-256"
+	automationAgentKeyFilePathInContainer = "/var/lib/mongodb-mms-automation/authentication/keyfile"
+	automationAgentWindowsKeyFilePath     = "%SystemDrive%\\MMSAutomation\\versions\\keyfile"
+	AgentName                             = "mms-automation"
+	AgentPasswordKey                      = "password"
+	AgentKeyfileKey                       = "keyfile"
+)
+
+type authEnabler struct {
+	agentPassword string
+	agentKeyFile  string
+}
+
+func (s authEnabler) EnableAuth(auth automationconfig.Auth) automationconfig.Auth {
+	s.enableAgentAuthentication(&auth)
```

Comment:
these 2 operations seem unbalanced, what do you think if you explicitely pass `s.agentPassword` and `s.agentKeyFile` to `enableAgentAuthentication` in order to make that function (`enableAgentAuthentication`) a regular function:

```go
enableAgentAuthentication(&auth, s.agentPassword, s.agentKeyFile)
enableDeploymentMechanisms(&auth)
return auth
```